temporal-sdk/src/main/java/io/temporal/internal/worker/ActivityPollTask.java

```diff
@@ -96,18 +98,19 @@ public ActivityTask poll() {
     PollActivityTaskQueueResponse response;
     SlotPermit permit;
     boolean isSuccessful = false;
-
+    CompletableFuture<SlotPermit> future =
```

Comment:
`reserveSlot` is a user defined function so I would except any use case to be under a `try` block so exceptions don't leek out and crash the worker or cause undesirable behaviour